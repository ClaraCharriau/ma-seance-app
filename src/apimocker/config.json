{
  "mockDirectory": "src/apimocker/",
  "quiet": false,
  "port": "7878",
  "latency": 50,
  "logRequestHeaders": false,
  "webServices": {
    "auth": {
      "mockFile": "users.json",
      "latency": 20,
      "verbs": ["post"]
    },
    "verify": {
      "mockFile": "verify.json",
      "latency": 20,
      "verbs": ["post"]
    }
  }
}
