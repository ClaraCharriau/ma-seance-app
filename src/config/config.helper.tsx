/* eslint-disable @typescript-eslint/no-implicit-any-catch */
/* 
    Set configuration file depending on the environment.
    dev.json is the default configuration file.
*/
let config = require('./dev.json');

if (process.env.REACT_APP_ENV === 'staging') {
    try {
        const stagingConfig = require('./staging.json');
        config = { ...config, ...stagingConfig };
    } catch (error) {
        console.error('Could not load staging configuration file', error);
    }
}

export default config;
