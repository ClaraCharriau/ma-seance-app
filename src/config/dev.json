{
    "services" : {
        "prefix": "http://localhost:7878",
        "auth": {
            "verify": "/verify",
            "login": "/login",
            "registrations": "/registrations"
        }
    },
    "tmdbImgPath" : {
        "medium": "https://image.tmdb.org/t/p/w500",
        "large": "https://image.tmdb.org/t/p/w1280"
    }
}