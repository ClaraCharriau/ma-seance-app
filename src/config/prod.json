{
    "services" : {
        "prefix-url": "http://localhost:7878"
    }
}