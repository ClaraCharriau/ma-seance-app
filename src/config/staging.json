{
    "services": {
        "auth": {
            "verify": "http://localhost:3030/maseance/api/v1/auth/verify",
            "login": "http://localhost:3030/maseance/api/v1/auth/login",
            "registrations": "http://localhost:3030/maseance/api/v1/auth/registrations"
        },
        "users": {
            "users": "http://localhost:3030/maseance/api/v1/users/",
            "favTheaters": "/favorite-theaters",
            "favMovies": "/watchlist-movies",
            "screenings": "/screenings"
        },
        "theaters": {
            "theaters": "http://localhost:8080/maseance/api/v1/theaters",
            "screenings": "/screenings"
        },
        "screenings": {
            "screenings": "/screenings"
        },
        "movies": {
            "movies": "http://localhost:8080/maseance/api/v1/movies",
            "currently": "/currently",
            "screenings": "/screenings"
        }
    }
}
