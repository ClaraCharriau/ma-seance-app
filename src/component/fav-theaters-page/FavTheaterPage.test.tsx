import { fireEvent, render, waitFor } from '@testing-library/react';
import FavTheaters from './FavTheaterPage';
import { BrowserRouter } from 'react-router-dom';
import mockFavoriteTheaters from '../../mocks/users/fav-theaters.json';
import mockUser from '../../mocks/users/users.json';
import { act } from 'react-dom/test-utils';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { axiosInstance } from '../../client/axios.config';

describe('Favorite theaters Component', () => {
    const favoriteContext = require('../../context/favorite.context');
    const authContext = require('../../context/auth.context');
    let axiosMock: MockAdapter;

    beforeEach(() => {
        axiosMock = new MockAdapter(axiosInstance);
    });
    afterEach(() => {
        axiosMock.reset();
    });
    it('should render favorite theaters component', () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockFavoriteTheaters
        });

        // When
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );

        // Then
        expect(component.container).toMatchSnapshot();
    });

    it('should render favorite theaters component when there are no favorite theaters', () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: []
        });

        // When
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );

        // Then
        expect(component.container).toMatchSnapshot();
    });

    it('should render favorite theaters delete buttons when user click on update button', () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockFavoriteTheaters
        });
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );

        // When
        fireEvent.click(component.getByText('modifier'));

        // Then
        waitFor(() => {
            expect(component.container).toMatchSnapshot();
        });
    });

    it('should render confirmation modale when clicking on delete button', async () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockFavoriteTheaters
        });
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );
        await act(() => {
            fireEvent.click(component.getByText('modifier'));
        });

        // When
        await act(() => {
            fireEvent.click(component.getAllByTestId('delete-button')[0]);
        });

        // Then
        await waitFor(() => {
            expect(
                component.getByText(
                    'Êtes-vous bien sûr de vouloir supprimer le cinéma C2L Saint-Germain de vos favoris ?'
                )
            ).toBeTruthy();
            expect(component.container).toMatchSnapshot();
        });
    });

    it('should call delete when user confirm deleting', async () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockFavoriteTheaters
        });
        jest.spyOn(authContext, 'useAuthContext').mockReturnValue({
            currentUser: mockUser
        });
        axiosMock.onDelete('/users/1/fav-theaters/1').reply(200);
        const axiosDelete = jest.spyOn(axiosInstance, 'delete');
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );
        await act(() => {
            fireEvent.click(component.getByText('modifier'));
        });
        await act(() => {
            fireEvent.click(component.getAllByTestId('delete-button')[0]);
        });

        // When
        await act(() => {
            fireEvent.click(component.getByText('confirmer'));
        });

        // Then
        await waitFor(() => {
            expect(component.container).toMatchSnapshot();
            expect(axiosDelete).toHaveBeenCalledWith('/users/1/fav-theaters/1');
        });
    });

    it('should close modal', async () => {
        // Given
        jest.spyOn(favoriteContext, 'useFavoriteContext').mockReturnValue({
            favoriteTheaters: mockFavoriteTheaters
        });
        const component = render(
            <BrowserRouter>
                <FavTheaters />
            </BrowserRouter>
        );
        await act(() => {
            fireEvent.click(component.getByText('modifier'));
        });
        await act(() => {
            fireEvent.click(component.getAllByTestId('delete-button')[0]);
        });

        // When
        await act(() => {
            fireEvent.click(component.getByText('annuler'));
        });

        // Then
        await waitFor(() => {
            expect(component.getByText("C2L Poissy")).toBeInTheDocument();
            expect(component.container).toMatchSnapshot();
        });
    });
});
