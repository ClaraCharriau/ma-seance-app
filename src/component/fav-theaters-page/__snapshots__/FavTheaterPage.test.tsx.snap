// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite theaters Component renders favorite theaters component 1`] = `
<div>
  <main
    class="mainFavTheaterPage"
  >
    <h2>
      Mes cinémas favoris
    </h2>
    <button
      class="updateButtonActive"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 12 12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 10.8118C0 11.4653 0.534653 11.9999 1.18812 11.9999H2.73267L0 9.26727V10.8118Z"
        />
        <path
          d="M11.6436 2.25743L9.74262 0.356436C9.26737 -0.118812 8.5545 -0.118812 8.07926 0.356436L7.96044 0.475247C7.72282 0.712871 7.72282 1.06931 7.96044 1.30693L10.6931 4.0396C10.9307 4.27723 11.2872 4.27723 11.5248 4.0396L11.6436 3.92079C12.1189 3.44554 12.1189 2.73267 11.6436 2.25743Z"
        />
        <path
          d="M7.1286 2.13861C6.89097 1.90098 6.53454 1.90098 6.29691 2.13861L0.890974 7.54455C0.593945 7.84158 0.593945 8.19801 0.831568 8.43563L3.56424 11.1683C3.80187 11.4059 4.1583 11.4059 4.39592 11.1683L9.80186 5.76237C10.0395 5.52474 10.0395 5.16831 9.80186 4.93069L7.1286 2.13861Z"
        />
      </svg>
      modifier
    </button>
    <section
      class="theaterList"
    >
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/1/day-1"
        >
          <img
            alt="photo de C2L Saint-Germain"
            class="theaterImg"
            src="/assets/img/theaters/c2l-saint-germain.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              C2L Saint-Germain
            </p>
            <p
              class="theaterAddress"
            >
              25-27-29, rue du Vieux-Marche 78100 Saint-Germain-en-Laye
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/2/day-1"
        >
          <img
            alt="photo de C2L Poissy"
            class="theaterImg"
            src="/assets/img/theaters/c2l-poissy.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              C2L Poissy
            </p>
            <p
              class="theaterAddress"
            >
              112 Rue du Général de Gaulle 78300 Poissy
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/3/day-1"
        >
          <img
            alt="photo de UGC Ciné Cité La Défense"
            class="theaterImg"
            src="/assets/img/theaters/ugc-defense-puteaux.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              UGC Ciné Cité La Défense
            </p>
            <p
              class="theaterAddress"
            >
              Le Dôme - Centre commercial Westfield - Les 4 Temps 92092 Puteaux
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/4/day-1"
        >
          <img
            alt="photo de Les 5 Caumartin"
            class="theaterImg"
            src="/assets/img/theaters/5-caumartin-paris.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              Les 5 Caumartin
            </p>
            <p
              class="theaterAddress"
            >
              101, rue Saint-Lazare 75009 Paris 9e arrondissement
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/5/day-1"
        >
          <img
            alt="photo de UGC Ciné Cité Les Halles"
            class="theaterImg"
            src="/assets/img/theaters/ugc-les-halles-paris.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              UGC Ciné Cité Les Halles
            </p>
            <p
              class="theaterAddress"
            >
              7 Place de la Rotonde 75001 Paris
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/6/day-1"
        >
          <img
            alt="photo de UGC Normandie"
            class="theaterImg"
            src="/assets/img/theaters/ugc-normandie-paris.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              UGC Normandie
            </p>
            <p
              class="theaterAddress"
            >
              116 bis Av. des Champs-Élysées 75008 Paris
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/7/day-1"
        >
          <img
            alt="photo de Publicis Cinémas"
            class="theaterImg"
            src="/assets/img/theaters/publicis-paris.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              Publicis Cinémas
            </p>
            <p
              class="theaterAddress"
            >
              129, av. des Champs-Elysées 75008 Paris 8e arrondissement
            </p>
          </div>
        </a>
      </div>
      <div
        class="theaterListCard"
      >
        <a
          class="theaterCardWrapper"
          href="/theaters/8/day-1"
        >
          <img
            alt="photo de Elysées Lincoln"
            class="theaterImg"
            src="/assets/img/theaters/lincoln-paris.jpg"
          />
          <div
            class="theaterCardInfos"
          >
            <p
              class="theaterName"
            >
              Elysées Lincoln
            </p>
            <p
              class="theaterAddress"
            >
              14 Rue Lincoln 75008 Paris
            </p>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`Favorite theaters Component renders favorite theaters component when there are no favorite theaters 1`] = `
<div>
  <main
    class="mainFavTheaterPage"
  >
    <h2>
      Mes cinémas favoris
    </h2>
    <button
      class="updateButtonActive"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 12 12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 10.8118C0 11.4653 0.534653 11.9999 1.18812 11.9999H2.73267L0 9.26727V10.8118Z"
        />
        <path
          d="M11.6436 2.25743L9.74262 0.356436C9.26737 -0.118812 8.5545 -0.118812 8.07926 0.356436L7.96044 0.475247C7.72282 0.712871 7.72282 1.06931 7.96044 1.30693L10.6931 4.0396C10.9307 4.27723 11.2872 4.27723 11.5248 4.0396L11.6436 3.92079C12.1189 3.44554 12.1189 2.73267 11.6436 2.25743Z"
        />
        <path
          d="M7.1286 2.13861C6.89097 1.90098 6.53454 1.90098 6.29691 2.13861L0.890974 7.54455C0.593945 7.84158 0.593945 8.19801 0.831568 8.43563L3.56424 11.1683C3.80187 11.4059 4.1583 11.4059 4.39592 11.1683L9.80186 5.76237C10.0395 5.52474 10.0395 5.16831 9.80186 4.93069L7.1286 2.13861Z"
        />
      </svg>
      modifier
    </button>
    <section
      class="theaterList"
    />
    <div
      class="emptySection"
    >
      <p>
        Vous n'avez pas encore de cinémas favoris
      </p>
      <button
        class="button"
      >
        Rechercher un 
        cinéma
      </button>
    </div>
  </main>
</div>
`;
